homeassistant/components/fibaro/sensor.py

```diff
@@ -64,7 +64,12 @@ async def async_setup_entry(
     """Set up the Fibaro controller devices."""
     entities: list[SensorEntity] = []
     for device in hass.data[DOMAIN][entry.entry_id][FIBARO_DEVICES]["sensor"]:
-        entities.append(FibaroSensor(device))
+        if device.type == "com.fibaro.energyMeter":
+            entities.append(FibaroEnergySensor(device))
```

Comment:
Unfortunately this is not enough as stated in my previous comment. The FibaroSensor takes his state from `self.fibaro_device.properties.value` whereas the FibaroEnergySensor takes his state from `self.fibaro_device.properties.energy`. So you can't solve the problem without changing either the FibaroSensor or the FibaroEnergySensor. If I can suggest something, I would add the SensorStateClass to SENSOR_TYPES array and then take this value up in FibaroSensor:
`SENSOR_TYPES = {
    "com.fibaro.temperatureSensor": [
        "Temperature",
        None,
        None,
        SensorDeviceClass.TEMPERATURE,
        SensorStateClass.MEASUREMENT,
    ],
    "com.fibaro.smokeSensor": [
        "Smoke",
        CONCENTRATION_PARTS_PER_MILLION,
        "mdi:fire",
        None,
        SensorStateClass.MEASUREMENT,
    ],
    "CO2": [
        "CO2",
        CONCENTRATION_PARTS_PER_MILLION,
        None,
        SensorDeviceClass.CO2,
        SensorStateClass.MEASUREMENT,
    ],
    "com.fibaro.humiditySensor": [
        "Humidity",
        PERCENTAGE,
        None,
        SensorDeviceClass.HUMIDITY,
        SensorStateClass.MEASUREMENT,
    ],
    "com.fibaro.lightSensor": [
        "Light",
        LIGHT_LUX,
        None,
        SensorDeviceClass.ILLUMINANCE,
        SensorStateClass.MEASUREMENT,
    ],
    "com.fibaro.energyMeter": [
        "Energy",
        ENERGY_KILO_WATT_HOUR,
        None,
        SensorDeviceClass.ENERGY,
        SensorStateClass.TOTAL_INCREASING,
    ],
}`

Then in initializing code you can just add the energy as other devices like this:
`for device in hass.data[DOMAIN][entry.entry_id][FIBARO_DEVICES][Platform.SENSOR]:
        entities.append(FibaroSensor(device))
        if "power" in device.interfaces:
            entities.append(FibaroPowerSensor(device))`

And in the device you assign the state class from SENSOR_TYPES structure:
`
        if fibaro_device.type in SENSOR_TYPES:
            self._unit = SENSOR_TYPES[fibaro_device.type][1]
            self._icon = SENSOR_TYPES[fibaro_device.type][2]
            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
            self._attr_state_class = SENSOR_TYPES[fibaro_device.type][4]
`